apps/swirl-docs/src/components/CodePreview/CodePreview.tsx

```diff
@@ -0,0 +1,122 @@
+import {
+  SwirlIconCheckStrong,
+  SwirlIconCopy,
+  SwirlIconExpandLess,
+  SwirlIconExpandMore,
+} from "@getflip/swirl-components-react";
+import { ReactNode, useState } from "react";
+import { CodeSandboxButton } from "./CodeSandboxButton";
+import { CopyToClipboard } from "react-copy-to-clipboard";
+import classNames from "classnames";
+import NoSsr from "../Layout/NoSsr";
+import Highlight, { defaultProps } from "prism-react-renderer";
+import theme from "prism-react-renderer/themes/vsDark";
+import { CodeExample } from "./types";
+import { NpmPackageLink } from "./NpmPackageLink";
+import CodePreviewContext from "./CodePreviewContext";
+import { Request } from "./CodePreviewRequestString";
+
+interface CodePreviewProps {
+  codeExample: CodeExample;
+  children?: ReactNode | ReactNode[];
+}
+
+/**
+ * Let's you easily render syntax highlighted code.
+ * Children of the Component will be displayed in the Header.
+ */
+export function CodePreview({ children, codeExample }: CodePreviewProps) {
+  const [isExpanded, setIsExpanded] = useState<boolean>(false);
+  const [isCopied, setIsCopied] = useState<boolean>(false);
+
+  return (
+    <NoSsr>
+      <CodePreviewContext.Provider value={codeExample}>
+        <div
+          className={classNames(
+            "relative w-auto bg-[#24292E] rounded-lg mb-10 overflow-auto",
+            {
+              "md:h-[240px] md:max-h-[240px] overflow-hidden": !isExpanded,
+              "min-h-[240px]": isExpanded,
+            }
+          )}
+        >
+          <div
+            className={classNames(
+              "hidden md:flex items-center justify-between h-12 m-2 p-4 rounded-lg",
+              "bg-[#21201E]"
+            )}
+          >
+            <div className="flex items-center justify-between">{children}</div>
+            <CopyToClipboard
+              text={codeExample.code}
+              onCopy={() => {
+                setIsCopied(true);
+                setTimeout(() => {
+                  setIsCopied(false);
+                }, 2000);
+              }}
+            >
+              <button className="flex justify-center items-center text-[#F2F2F2] text-base font-medium">
+                {isCopied ? "copied!" : "copy"}
```

Comment:
**nitpick:** "copy" is a bit weak as a label. I think "Copy code" and "Code copied." would be more concise.